diff -Naur proftpd-1.3.4e/config.guess proftpd-1.3.4e-patched/config.guess
--- proftpd-1.3.4e/config.guess	2012-05-25 18:47:44.000000000 +0200
+++ proftpd-1.3.4e-patched/config.guess	2017-01-31 20:58:00.782104383 +0100
@@ -898,6 +898,9 @@
 	    fi
 	fi
 	exit ;;
+    arc*:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit 0 ;;
     avr32*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
