diff -Naur ipsec-tools-0.7.2/src/racoon/isakmp.c ipsec-tools-0.7.2-patched/src/racoon/isakmp.c
--- ipsec-tools-0.7.2/src/racoon/isakmp.c	2009-01-19 19:49:32.000000000 +0100
+++ ipsec-tools-0.7.2-patched/src/racoon/isakmp.c	2017-02-12 11:25:28.442015912 +0100
@@ -110,7 +110,7 @@
 #ifdef ENABLE_NATT
 # include "nattraversal.h"
 #endif
-# ifdef __linux__
+# if defined(__linux__) || defined(__arc__)
 #  include <linux/udp.h>
 #  include <linux/ip.h>
 #  ifndef SOL_UDP
@@ -199,7 +199,7 @@
 		char		buf[sizeof (isakmp) + 4];
 		u_int32_t	non_esp[2];
 		char		lbuf[sizeof(struct udphdr) + 
-#ifdef __linux
+# if defined(__linux) || defined(__arc__)
 				     sizeof(struct iphdr) + 
 #else
 				     sizeof(struct ip) + 
@@ -241,7 +241,7 @@
 	/* Lucent IKE in UDP encapsulation */
 	{
 		struct udphdr *udp;
-#ifdef __linux__
+# if defined(__linux__) || defined(__arc__)
 		struct iphdr *ip;
 
 		udp = (struct udphdr *)&x.lbuf[0];
@@ -1633,7 +1633,7 @@
 		switch (p->addr->sa_family) {
 		case AF_INET:
 			if (setsockopt(p->sock, IPPROTO_IP,
-#ifdef __linux__
+# if defined(__linux__) || defined(__arc__)
 				       IP_PKTINFO,
 #else
 				       IP_RECVDSTADDR,
